### ISSUE TITLE
Make Java generic methods children of abs nodes [phab:D1135]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1494]
_Created by **salguarnieri** at 2016-11-04 01:36:30_: 

Make wildcard nodes params of the abs node created for a generic method. This means we will also start creating abs nodes for methods that don't have any named type parameters but do use wildcards in at least one of the types for the parameters.

**Comment by kythe** [comment]
_**salguarnieri** wrote at 2016-11-04 23:28:05_:

- rebased

**Comment by kythe** [comment]
_**zarko** wrote at 2016-11-04 23:39:34_:

It looks like a call to wildcard in WildcardMethod points at WildcardFnDecl and not WildcardFnAbs:

```
  private static void call() {
    //- wildcard ref What?
    //- "wildcard(null)" ref/call Whom?
    wildcard(null);
  }
```

```
WildcardFnAbs: EVar(0x0000000000e60298 = App(vname, (d303488ec629c9e62a16988b5234f21f24acc31277aafdbea5b1fd61727b732a, "", "", "", java)))
WildcardFnDecl: EVar(0x0000000000e606d8 = App(vname, (d5aa5eec2c5d99ad4d63fc4e0b2b70386fd308f78b16a588d9a13f749e07ccf8, kythe, "", kythe/javatests/com/google/devtools/kythe/analyzers/java/testdata/pkg/WildcardMethod.java, java)))
What: EVar(0x0000000000e61188 = App(vname, (d5aa5eec2c5d99ad4d63fc4e0b2b70386fd308f78b16a588d9a13f749e07ccf8, kythe, "", kythe/javatests/com/google/devtools/kythe/analyzers/java/testdata/pkg/WildcardMethod.java, java)))
Whom: EVar(0x0000000000e612b0 = App(vname, (d5aa5eec2c5d99ad4d63fc4e0b2b70386fd308f78b16a588d9a13f749e07ccf8, kythe, "", kythe/javatests/com/google/devtools/kythe/analyzers/java/testdata/pkg/WildcardMethod.java, java)))
```

We'd discussed leaving out the intervening `tapp(WildcardFnAbs, Object)`, but clients will expect to see something with `%defines/binding` edges at the other end of a `ref*` edge; unless there's a good reason to do otherwise, edges pointing to nodes covered by `abs` nodes should connect to the `abs` node.

(also, why is the `kythe` corpus missing from WildcardFnAbs's VName?)

**Comment by kythe** [comment]
_**salguarnieri** wrote at 2016-11-07 22:54:52_:

  - Add some simple tests and a todo for future improvement

**Comment by kythe** [comment]
_**salguarnieri** wrote at 2016-11-07 22:54:59_:

We discussed a bit offline and I added failing tests and #1501 for followup work.

**Comment by kythe** [comment]
_**zarko** accepted this revision at 2016-11-08 00:18:25_

**Comment by kythe** [comment]
_**salguarnieri** closed via commit 0264461a8dec2d7877ca2243a1cfff0e06bc262a at 2016-11-08 00:21:58_
